src/kaleidoscope/device/dygma/Raise.cpp

```diff
@@ -0,0 +1,599 @@
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef ARDUINO_SAMD_RAISE
+
+#include <Kaleidoscope.h>
+#include <Kaleidoscope-EEPROM-Settings.h>
+#include <Kaleidoscope-LEDControl.h>
+#include <KeyboardioHID.h>
+#include <Wire.h>
+
+#include "kaleidoscope/util/crc16.h"
+
+#define I2C_CLOCK_KHZ 200
+#define I2C_FLASH_CLOCK_KHZ 100 // flashing doesn't work reliably at higher clock speeds
+
+#define SIDE_POWER 1 // side power switch pa10
+
+#define LKEY 33 // raise::left_keys
+#define LPH LEDS_PER_HAND
+#define XX 0xFF // off
+
+const uint8_t PROGMEM gamma8[] = {
+  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,  1,  1,  1,
+  1,  1,  1,  1,  1,  1,  1,  1,  1,  2,  2,  2,  2,  2,  2,  2,
+  2,  3,  3,  3,  3,  3,  3,  3,  4,  4,  4,  4,  4,  5,  5,  5,
+  5,  6,  6,  6,  6,  7,  7,  7,  7,  8,  8,  8,  9,  9,  9, 10,
+  10, 10, 11, 11, 11, 12, 12, 13, 13, 13, 14, 14, 15, 15, 16, 16,
+  17, 17, 18, 18, 19, 19, 20, 20, 21, 21, 22, 22, 23, 24, 24, 25,
+  25, 26, 27, 27, 28, 29, 29, 30, 31, 32, 32, 33, 34, 35, 35, 36,
+  37, 38, 39, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 50,
+  51, 52, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 66, 67, 68,
+  69, 70, 72, 73, 74, 75, 77, 78, 79, 81, 82, 83, 85, 86, 87, 89,
+  90, 92, 93, 95, 96, 98, 99, 101, 102, 104, 105, 107, 109, 110, 112, 114,
+  115, 117, 119, 120, 122, 124, 126, 127, 129, 131, 133, 135, 137, 138, 140, 142,
+  144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 167, 169, 171, 173, 175,
+  177, 180, 182, 184, 186, 189, 191, 193, 196, 198, 200, 203, 205, 208, 210, 213,
+  215, 218, 220, 223, 225, 228, 231, 233, 236, 239, 241, 244, 247, 249, 252, 255
+};
+
+// these are zero indexed led numbers from sled matrix
+// maps rows and columns to the keyboard led number (0->PROPS_::LED_COUNT-1)
+// 19 is missing for ANSI
+static constexpr uint8_t key_led_map[5][16] = {
+  // ISO & ANSI (ANSI has no LED at 20, but this key can never be pressed so we can have just one map).
+  {0,  1,  2,  3,  4,  5,  6,  XX,      XX,   6 + LKEY, 5 + LKEY, 4 + LKEY, 3 + LKEY, 2 + LKEY, 1 + LKEY, 0 + LKEY},
+  {7,  8,  9,  10, 11, 12, XX, XX,      14 + LKEY, 13 + LKEY, 12 + LKEY, 11 + LKEY, 10 + LKEY, 9 + LKEY, 8 + LKEY, 7 + LKEY},
+  {13, 14, 15, 16, 17, 18, XX, XX,      XX,   21 + LKEY, 20 + LKEY, 19 + LKEY, 18 + LKEY, 17 + LKEY, 16 + LKEY, 15 + LKEY},
+  {19, 20, 21, 22, 23, 24, 25, XX,      XX, XX,   27 + LKEY, 26 + LKEY, 25 + LKEY, 24 + LKEY, 23 + LKEY, 22 + LKEY},
+  {26, 27, 28, 29, 30, XX, 31, 32,      35 + LKEY, 34 + LKEY, 33 + LKEY, 32 + LKEY, 31 + LKEY, 30 + LKEY, 29 + LKEY, 28 + LKEY},
+};
+
+// maps keyboard led number (0->PROPS_::LED_COUNT-1) to the SLED led number (0->LPH-1 on left side  and LPH to LPH*2-1 on the right side)
+// LPH comes from keyboardioscanner.h - leds per hand = 72 defined by the size of the buffer used to transfer data to sides
+static constexpr uint8_t led_map[2][133] = {
```

Comment:
The 2 and 133 in the led_map are slightly opaque magic numbers. The code could be improved by breaking them out to named magic numbers at least.